plugins/module_utils/cloudfront_facts.py

```diff
@@ -26,109 +26,137 @@
 Common cloudfront facts shared between modules
 """
 
+from functools import partial
 try:
     import botocore
 except ImportError:
     pass
 
 from .ec2 import AWSRetry
 from .ec2 import boto3_tag_list_to_ansible_dict
+from .ec2 import snake_dict_to_camel_dict
 
 
-class CloudFrontFactsServiceManager:
-    """Handles CloudFront Facts Services"""
+class CloudFrontFactsServiceManagerFailure(Exception):
+    pass
 
-    def __init__(self, module):
-        self.module = module
-        self.client = module.client('cloudfront', retry_decorator=AWSRetry.jittered_backoff())
 
-    def get_distribution(self, distribution_id):
-        try:
-            return self.client.get_distribution(Id=distribution_id, aws_retry=True)
-        except botocore.exceptions.ClientError as e:
-            self.module.fail_json_aws(e, msg="Error describing distribution")
+def cloudfront_facts_keyed_list_helper(list_to_key):
+    keyed_list = dict()
+    for item in list_to_key:
+        distribution_id = item['Id']
+        if 'Items' in item['Aliases']:
+            aliases = item['Aliases']['Items']
+            for alias in aliases:
+                keyed_list.update({alias: item})
```

Comment:
I don't why you are saying that it sounds wrong, basically the code extend the item so that we can find element using either distribution id or one its aliases